docs/source/developers/java/building.rst

```diff
@@ -0,0 +1,239 @@
+    $ cd java-native-cpp
+    $ cmake \
+        -DARROW_BOOST_USE_SHARED=OFF \
+        -DARROW_BROTLI_USE_SHARED=OFF \
+        -DARROW_BZ2_USE_SHARED=OFF \
+        -DARROW_GFLAGS_USE_SHARED=OFF \
+        -DARROW_GRPC_USE_SHARED=OFF \
+        -DARROW_LZ4_USE_SHARED=OFF \
+        -DARROW_OPENSSL_USE_SHARED=OFF \
+        -DARROW_PROTOBUF_USE_SHARED=OFF \
+        -DARROW_SNAPPY_USE_SHARED=OFF \
+        -DARROW_THRIFT_USE_SHARED=OFF \
+        -DARROW_UTF8PROC_USE_SHARED=OFF \
+        -DARROW_ZSTD_USE_SHARED=OFF \
+        -DARROW_JNI=ON \
+        -DARROW_PARQUET=ON \
+        -DARROW_FILESYSTEM=ON \
+        -DARROW_DATASET=ON \
+        -DARROW_GANDIVA_JAVA=ON \
+        -DARROW_GANDIVA_STATIC_LIBSTDCPP=ON \
+        -DARROW_GANDIVA=ON \
+        -DARROW_ORC=ON \
+        -DARROW_PLASMA_JAVA_CLIENT=ON \
+        -DARROW_PLASMA=ON \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_LIBDIR=lib \
+        -DCMAKE_INSTALL_PREFIX=../java-dist \
+        -DCMAKE_UNITY_BUILD=ON \
+        -Dre2_SOURCE=BUNDLED \
+        -DBoost_SOURCE=BUNDLED \
+        -Dutf8proc_SOURCE=BUNDLED \
+        -DSnappy_SOURCE=BUNDLED \
+        -DORC_SOURCE=BUNDLED \
+        -DZLIB_SOURCE=BUNDLED \
+        ../cpp
+    $ cmake --build . --target install
+    $ ls -latr  ../java-dist/lib
+    |__ libarrow_dataset_jni.dylib
+    |__ libarrow_orc_jni.dylib
+    |__ libgandiva_jni.dylib
+
+Building Arrow JNI Modules
+--------------------------
+
+To compile the JNI bindings, use the ``arrow-c-data`` Maven profile:
+
+.. code-block::
+
+    $ cd arrow/java
+    $ mvn -Darrow.c.jni.dist.dir=../java-dist/lib -Parrow-c-data clean install
+
+To compile the JNI bindings for ORC / Gandiva / Dataset, use the ``arrow-jni`` Maven profile:
+
+.. code-block::
+
+    $ cd arrow/java
+    $ mvn -Darrow.cpp.build.dir=../java-dist/lib -Parrow-jni clean install
+
+IDE Configuration
+=================
+
+IntelliJ IDE
+------------
+
+Go to open java project and select java folder.
+
+When IntelliJ open arrow java project its compiled automatically and show you maven modules inside root project and
+also offer a side bar to see profile and maven plugin configured by default.
+
+.. figure:: img/java_welcome.png
+   :scale: 50 %
+   :alt: A newly opened Arrow project
+
+Also IntelliJ offer options to configure custom run/debug configurations to setup maven module working directory,
+maven command line to run, profiles, configure JRE to use, add environment variables as some examples of these.
+Option available at: Run -> Edit Configurations.
+
+Let's create our maven configuration: java-dataset
+
+.. figure:: img/java_mvn_configuration.png
+   :scale: 50 %
+   :alt: Setup new maven configurations with working directory, command to run and profile used to build
+
+Let's define our JRE runner:
+
+.. figure:: img/java_jre_runner.png
+   :scale: 50 %
+   :alt: Setup maven configurations with JRE selected
+
+Let's configure additional environment variables:
```

Comment:
This is only necessary if we want to work with the JNI modules, right? We should explain that.

Reply:
Deleted because is adding complexity to simple task